File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/tools/ToolsSettings.java
package com.github.alexandrecarlton.idea.settings.layout.tools;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.github.alexandrecarlton.idea.settings.layout.tools.file_watchers.FileWatcherSettings;
import org.immutables.value.Value;

import java.util.List;

@Value.Immutable
@JsonDeserialize(as = ImmutableToolsSettings.class)
public interface ToolsSettings {

  List<FileWatcherSettings> fileWatchers();

}

File: src/main/kotlin/com/github/alexandrecarlton/idea/settings/layout/tools/ToolsSettings.kt
package com.github.alexandrecarlton.idea.settings.layout.tools

import com.fasterxml.jackson.annotation.JsonProperty
import com.github.alexandrecarlton.idea.settings.layout.tools.file_watchers.FileWatcherSettings

data class ToolsSettings(
    @JsonProperty("File Watchers")
    val fileWatchers: List<FileWatcherSettings>? = null
)
